splice_machine/src/main/java/com/splicemachine/derby/impl/stats/StoreCostControllerImpl.java

```diff
@@ -183,6 +191,18 @@ public StoreCostControllerImpl(TableDescriptor td, ConglomerateDescriptor conglo
             tableStatistics = new TableStatisticsImpl(tableId, partitionStats,fallbackNullFraction,extraQualifierMultiplier);
             useRealTableStatistics = true;
         }
+
+        long tableSize = tableStatistics.rowCount() * tableStatistics.avgRowWidth();
+        if (isMemPlatform())
+            parallelism = 1;
+        else {
+            if (requestedSplits > 0)
+                parallelism = requestedSplits;
+            else
+                parallelism = EngineDriver.getNumSplits(tableSize, getNumPartitions());
+            if (parallelism > getMaxExecutorCores())
+                parallelism = getMaxExecutorCores();
+        }
```

Comment:
I'm concerned about the impact of the call for getNumSplits() for OLTP queries. Now that we've separated the costing for OLTP and OLAP in two passes, I think we can let the OLTP also goes through the IF path here, that is, parallelism =1. We can trigger the computation of the number of splits only in the pass of OLAP costing.

Reply:
The StoreCostController for a table is a fixed entity that should only be built once per query (ideally).  Think of the field "parallelism" really as "parallelismIfWeRunOnSpark".  The value of parallelism returned by SimpleCostEstimate.getParallelism is always 1, unless we are in the isForSpark pass:

>     @Override public int getParallelism() {
>         if (optimizer == null || optimizer.isForSpark())
>             return parallelism;
>         else
>             return 1;
>     }

The reason I have to special-case isMemPlatform here, is that even though we might do a spark pass on mem platform, we actually can never send the query to spark when running mem, therefore it is a fake spark pass, and we should force parallelism to 1.

If you think the name "parallelism" is not clear, I could rename it to "parallelismIfWeRunOnSpark", or something else.

Reply:
I got the wrong impression that the call of EngineDriver.getNumSplits() is expensive, so I had the concern that OLTP query will pay the price to call EngineDriver.getNumSplits() even though OLTP costing does not need it. Now looking at this function again, it is not really expensive, so my concern is gone. 
There is no need to rename, thanks!

Reply:
OK, thanks.